src/persist-txn/src/txn_cache.rs

```diff
@@ -970,18 +1002,76 @@ impl<T: Timestamp + TotalOrder> DataTimes<T> {
 
 #[derive(Debug)]
 pub(crate) enum Unapplied<'a> {
-    Batch(&'a Vec<u8>),
     Register,
+    Batch(&'a Vec<u8>),
+    Forget,
+}
+
+#[derive(Debug)]
+struct UnappliedIterInput<'a, T>
+where
+    T: Timestamp + Lattice + TotalOrder + StepForward + Codec64,
+{
+    unapplieds: VecDeque<(&'a ShardId, Unapplied<'a>, &'a T)>,
```

Comment:
There might be a way to structure this so we don't need to allocate a new `VecDeque` for each input, but I couldn't figure it out and this is probably fine.

Reply:
Why'd we have to introduce it? Because we went from multiplexing 2 things to 3 things?

Reply:
Yeah, pretty much. Before we had to allocate a `Vec` for the unapplied writes and keep a reference to the `VecDeque` of unapplied registers. Since we went from 2 to 3 things I wanted something generic that we could iterate over. I would have much preferred something like
```Rust
unapplied: impl Peekable<Item = (&'a ShardId, Unapplied<'a>, &'a T)>
```
so we wouldn't have to allocate a `Vec`/`VecDeque` for any of the inputs, but I never got that to work.

Reply:
> I just pushed an update to combine them. I kept around the updated unapplied iterator to make it easier to separate them in the future, let me know if you'd prefer I revert that.

Yeah, I think I preferred the old impl (reads more obviously to me and it's without the extra VecDeques) 🙈

Reply:
Done